File: finkzeit/finkzeit/report/uva_details/uva_details.py
# Copyright (c) 2019, Fink Zeitsysteme/libracore and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe

def execute(filters=None):
    columns, data = [], []
    
    # prepare columns
    columns = ["Datum::100", 
        "Beleg::100", 
        "Basis:Currency:100", 
        "Netto:Currency:100", 
        "Steuerregel::200", 
        "Steuer:Currency:100"
    ]
    
    # prepare filters
    if filters.code:
        view = "viewATVAT_{0}".format(filters.code)
    else:
        view = "viewATVAT_000"
    if filters.from_date:
        from_date = filters.from_date
    else:
        from_date = "2000-01-01"
    if filters.to_date:
        to_date = filters.to_date
    else:
        to_date = "2999-12-31"
        
    # VAT control view
    sql_query = """SELECT *
                FROM `{view}`
                WHERE 
                  `{view}`.`posting_date` >= '{from_date}'
                  AND `{view}`.`posting_date` <= '{to_date}';""".format(view=view, from_date=from_date, to_date=to_date)
    data = frappe.db.sql(sql_query, as_list = True)
    
    # return data
    return columns, data

File: finkzeit/finkzeit/report/uva_details/uva_details.js
// Copyright (c) 2016, Fink Zeitsysteme/libracore and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["UVA Details"] = {
	"filters": [
        {
            "fieldname":"code",
            "label": __("Code"),
            "fieldtype": "Select",
            "options": "000\n011\n017\n021\n022\n029\n057\n060\n061\n065\n066\n070\n072\083",
            "default": "000"
        },
        {
            "fieldname":"from_date",
            "label": __("From Date"),
            "fieldtype": "Date",
            "default": get_one_month_ago(new Date())
        },
        {
            "fieldname":"to_date",
            "label": __("To Date"),
            "fieldtype": "Date",
            "default": new Date()
        }
	]
}

// returns the date one month ago
function get_one_month_ago(date) {
    d = date;
    d.setMonth(d.getMonth() - 1);
    return d;
}
